Updated writers to work with latest version of Foyer

summeraz opened the pull request:
Slightly different syntax required to apply a forcefield to a Compound.  Also fixed a bug in the LAMMPS writer where an error would occur if a Compound lacked bonds, angles, or dihedrals.

Unit tests for the writers are broken with the new Foyer version, as the path to the force field file is not specified.  Any thoughts on the best way to handle this?  We could have Foyer search the "forcefields" directory by default for matching force field names, rather than needing to specify the full path to the XML file.

ctk3b commented:
That is exactly what the `by_name` method should be doing. What exactly is the error?

summeraz commented:
Oops, I thought I was using `by_name` but apparently not.  That should fix the issue

summeraz commented:
Oh now I see what happened.  I had been using `by_name`, but that only works for a single force field.  I switched to using the constructor directly so that multiple force field names could be provided to combine.  I've changed it now so that `by_name` is used if only a single string is provided, and if multiple strings are provided in a list then the constructor is called directly (and the full path to the XML file must be supplied in each string).  I can edit the `by_name` function in foyer to accept multiple strings to handle this properly.
